superset-frontend/src/cccs-viz/plugins/plugin-chart-at-a-glance-ip/src/AtAGlanceIP.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useState } from 'react';
+import { RiGlobalFill } from 'react-icons/ri';
+import { QueryFormData } from '@superset-ui/core';
+// import styles from './styles';
+
+const DEFAULT_IP_STRING = '1.1.1.1';
+const IP_STRING = 'ip_string';
+
+const setInitialIpString = (adhocFilters: any) => {
+  let ipString: string = DEFAULT_IP_STRING;
+  if (adhocFilters instanceof Array && adhocFilters.length > 0) {
+    const adhocFilter = adhocFilters[0];
+    if (
+      adhocFilter.comparator instanceof Array &&
+      adhocFilter.comparator.length > 0
+    ) {
+      ipString = adhocFilter.comparator[0];
+    } else if (adhocFilter.comparator instanceof String) {
+      ipString = adhocFilter.comparator;
+    }
+  }
+  return ipString;
+};
```

Comment:
Probably but then your plugins are going to have a dependency outside of the individual plugin itself.  It depends on whether or not we want our UI plugins to be truly independent "plugins" or not.  For now all the code needed for the plugin is inside of the plugin.  I tossed this idea around as I was separating things out and rather than prolonging the work I went with this bit of repetition.

Reply:
That could potentially be tackled with the beautify task, should we really want to get rid of this bit of redundancy.